packages/siop-oid4vp/lib/authorization-response/AuthorizationResponse.ts

```diff
@@ -185,7 +207,8 @@ export class AuthorizationResponse {
       state,
       correlationId: verifyOpts.correlationId,
       ...(this.idToken && { idToken: verifiedIdToken }),
-      ...(oid4vp && { oid4vpSubmission: oid4vp }),
+      ...(oid4vp && 'presentationDefinitions' in oid4vp && { oid4vpSubmission: oid4vp }),
+      ...(oid4vp && 'dcqlQuery' in oid4vp && { oid4vpSubmissionDcql: oid4vp }),
```

Comment:
maybe in `oid4vp` add a prop `queryType` with either value `dcql` or `presentationDefinition`/`pex`?  Then the type of oid4vp can be defined afterwards. I'm not a fan of the `in` syntax for type narrowing.